src/Stack/Types/Build.hs

```diff
@@ -142,7 +142,7 @@ instance Show StackBuildException where
                ", the package id couldn't be found " <> "(via ghc-pkg describe " <>
                packageNameString name <> "). This shouldn't happen, " <>
                "please report as a bug"
-    show (CompilerVersionMismatch mactual (expected, earch) ghcVariant ghcBuild check mstack resolution) = concat
+    show (CompilerVersionMismatch mactual (expected, eArch) ghcVariant ghcBuild check mstack resolution) = concat
```

Comment:
There seems to be some willingness to capitalize for words,. I _think_ that this is `e`+`arch`, and not a typo for `search`. Note that I'm not doing things to `m`+_words_ (e.g. `mstack`) but, here it seemed harmless.

Reply:
I am fine with this sort of refactoring and hopefully the CI will pick up if anything has been 'broken' by the change. However, it is difficult to review a mix of refactoring with a large volume of documentation/output fixes.